### PR TITLE
Osx threadfix

### PR DESCRIPTION
**Pull request by nhorman**
test build for debugging, as local operation on the development m1 doesn't reproduce failure

### CODE REVIEW

**Comment by slontis** [comment]
Is there some way of limiting the number of writes instead of doing this?

**Comment by nhorman** [comment]
sure, but that runs a bit counter to the purpose of the test, they idea (in part) was to not only test the sanity of rw and rcu locks, but to provide a level of performance comparison.

What we could do is, instead of wrapping, when we hit MAX_ULLONG, just leave the counter there so that we never tripped over the wrapping case.  Would that be acceptable?

**Comment by t-j-h** [comment]
If you are running over the end of the unsigned long long then you need to implement a "carry" into another variable (perhaps even another unsigned long long) each time you reach the wrapping point. That does sound like overkill for this context - but I also find the implementation here more than a little strange.

**Comment by t8m** [comment]
@nhorman I do not believe a wrap around 2^64 bit value is happening here. Or, the starting point was not a zero value. If the test run for 1000 seconds (which it is not running) it would mean doing 18 milions iterations in every nanosecond - that's obviously impossible. There must be some other issue in play.

**Comment by kroeckx** [comment]
If we're worried for a wrap around at 2^64, why not clear the first bit?

File: test/threadstest.c [hunk 1]
```diff
@@ -1223,8 +1225,10 @@ int setup_tests(void)
 #if defined(OPENSSL_THREADS)
     ADD_TEST(torture_rw_low);
     ADD_TEST(torture_rw_high);
+#ifndef OPENSSL_SYS_MACOSX
```
Comment (t8m) [inline test/threadstest.c:1228]: Nit: space after `#` - applies to all other places where `#ifdef/#endif` was added

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by nhorman** [comment]
Based on OTC discussion, for the purposes of the beta release, since we are not yet using rcu, I've updated this PR to disable the rcu torture test on macosx.  Will re-enable prior to introduction of usage post release

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
OK with urgent

File: test/threadstest.c [hunk 2]
```diff
@@ -292,7 +293,6 @@ static int writer2_iterations = 0;
 static uint64_t *writer_ptr = NULL;
 static uint64_t global_ctr = 0;
 static int rcu_torture_result = 1;
-
```
Comment (mattcaswell) [inline test/threadstest.c:295]: This blank line removal looks odd. In my mind it is better with it.

**Review by mattcaswell — APPROVED** [review]
Approved. I have one minor comment - but approved either way.

Agree urgent.

**Comment by nhorman** [comment]
merged to master and 3.3

**Pull request closed without merging**